rskj-core/src/main/java/co/rsk/cli/RskCli.java

```diff
@@ -76,6 +77,12 @@ static class NetworkFlags {
     @CommandLine.Option(names = {"-X"}, description = "Read arguments in command line")
     private List<String> xArguments;
 
+    @CommandLine.Option(names = {"--sync-mode"}, description = "Set Synchronization mode. Valid options are <full | snap>")
+    private String syncMode;
```

Comment:
I guess you could use `SyncMode` instead of `String`, so you won't need to parse it manually

Reply:
Because picocli is case-sensitive and setting up a custom converter seems like overkill for this case, it’s simpler to stick with syncMode as a String.

This way, we handle case insensitivity by converting the string to uppercase before processing it. This approach will handle any case variations like full, FULL, or Snap correctly:

```
SYNC_MODE("sync-mode", true) {
    @Override
    public Config withConfig(Config config, String configValue) {
        SyncMode mode;
        try {
            mode = SyncMode.valueOf(configValue.toUpperCase());
        } catch (IllegalArgumentException e) {
            throw new IllegalArgumentException("Invalid sync mode: " + configValue + ". Valid options are <full> or <snap>.");
        }

        if (mode == SyncMode.SNAP) {
            return config.withValue(RskSystemProperties.PROPERTY_SNAP_CLIENT_ENABLED, ConfigValueFactory.fromAnyRef(true));
        }
        return config;
    }
}
```
Using the SyncMode enum directly would require a custom converter to handle case insensitivity, which seems like extra work for our needs.

Let me know if this approach works for you or if you have any other thoughts!